test/rhsmlib_test/test_pkg_arches.py

```diff
@@ -0,0 +1,59 @@
+try:
+    import unittest2 as unittest
+except ImportError:
+    import unittest
+
+from rhsmlib.facts import pkg_arches
+from mock import patch
+
+
+class TestSupportedArchesCollector(unittest.TestCase):
+
+    def setUp(self):
+        self.collector = pkg_arches.SupportedArchesCollector()
+
+    @patch('platform.linux_distribution')
+    @patch('subprocess.check_output')
+    def test_single_arch_on_ubuntu(self, mock_check_output, mock_linux_distribution):
+
+        def single(*args, **kwargs):
+            if args[0] == 'dpkg --print-architecture':
+                return 'amd64\n'
+            return ''
+
+        mock_linux_distribution.return_value = ("Ubuntu", None, None)
+        mock_check_output.side_effect = single
+        fact = self.collector.get_all()
+        self.assertEqual(fact["supported_architectures"], ['amd64'])
+
+    @patch('platform.linux_distribution')
+    @patch('subprocess.check_output')
+    def test_multi_arch_on_ubuntu(self, mock_check_output, mock_linux_distribution):
+
+        def multi(*args, **kwargs):
+            if args[0] == 'dpkg --print-architecture':
+                return 'amd64\n'
+            elif args[0] == 'dpkg --print-foreign-architectures':
+                return 'i386\n'
+            return ''
+
+        mock_linux_distribution.return_value = ("Debian", None, None)
+        mock_check_output.side_effect = multi
+        fact = self.collector.get_all()
+        self.assertEqual(fact["supported_architectures"], ['amd64', 'i386'])
+
+    @patch('platform.linux_distribution')
+    @patch('subprocess.check_output')
+    def test_none_arch_on_redhat(self, mock_check_output, mock_linux_distribution):
+
+        def multi(*args, **kwargs):
+            if args[0] == 'dpkg --print-architecture':
+                return 'amd64\n'
+            elif args[0] == 'dpkg --print-foreign-architectures':
+                return 'i386'
+            return ''
+
+        mock_linux_distribution.return_value = ("RedHat", None, None)
+        mock_check_output.side_effect = multi
+        fact = self.collector.get_all()
+        self.assertIsNone(fact)
```

Comment:
This assertion is wrong. There should be something like `self.assertEqual(fact, {})`. Unit tests are failing due to this asseration.

Reply:
Done. sorry to bother you again.